lib/matplotlib/tests/test_colorbar.py

```diff
@@ -1230,3 +1230,8 @@ def test_colorbar_wrong_figure():
     fig_tl.colorbar(im)
     fig_tl.draw_without_rendering()
     fig_cl.draw_without_rendering()
+
+
+def test_colorbar_format_string_and_old():
+    plt.imshow([[0, 1]])
+    plt.colorbar(format="{x}%")
```

Comment:
Let's not only test that this does not raise anymore, but also that it detects the correct formatter:
```suggestion
    cb = plt.colorbar(format="{x}%")
    assert isinstance(cb._formatter, StrMethodFormatter)
```
(you may need to import `StrMethodFormatter`)